### DISCUSSION TITLE
Pull git repo with submodule

### DISCUSSION THREAD

**Discussion by ramazulay** [D10S0VSkY-OSS/Stack-Lifecycle-Deployment#70] [Ideas]
Hi @D10S0VSkY-OSS,
There is a way to pull the git repo with submodule?

Thanks.

**Comment by D10S0VSkY-OSS** [comment]
Hi @ramazulay ,
I haven't tried it yet...
But I'll check it out and let you know ASAP
Regards

**Reply by ramazulay** [reply]
Thanks a lot.

**Comment by D10S0VSkY-OSS** [comment]
Hi @ramazulay ,
included in the latest version  v2.5.1
Regards

**Reply by ramazulay** [reply]
Hi @D10S0VSkY-OSS,
Are you sure the image exist?

Normal   BackOff    20s               kubelet            Back-off pulling image "d10s0vsky/sld-api:2.5.1"
  Warning  Failed     20s               kubelet            Error: ImagePullBackOff
  Normal   Pulling    9s (x2 over 22s)  kubelet            Pulling image "d10s0vsky/sld-api:2.5.1"
  Warning  Failed     7s (x2 over 21s)  kubelet            Failed to pull image "d10s0vsky/sld-api:2.5.1": rpc error: code = Unknown desc = Error response from daemon: manifest for d10s0vsky/sld-api:2.5.1 not found: manifest unknown: manifest unknown
  Warning  Failed     7s (x2 over 21s)  kubelet            Error: ErrImagePull

**Comment by D10S0VSkY-OSS** [comment]
Hi @ramazulay ,
v2.5.1 is the code version SLD no image docker
**d10s0vsky/sld-api:2.5.0**
**d10s0vsky/sld-dashboard:2.4.1**

1. Create terraform stack git repo
2. Add git module ( git submodule add )
3. SLD now pull git module as a folder with same name git module repo

**Reply by ramazulay** [reply]
Hi @D10S0VSkY-OSS,
Ohh, thanks.
When we resolve the [issue](https://github.com/D10S0VSkY-OSS/Stack-Lifecycle-Deployment/issues/75) I will be create/update the "stack".

Thanks.

**Comment by ramazulay** [comment]
Hi @D10S0VSkY-OSS,
Can you add --recursive to the "git clone" command.
Our repository must be submodule for terraform deployment.

Thanks.

**Reply by D10S0VSkY-OSS** [reply]
Hi @ramazulay ,
recursion already does.
What problem do you have exactly, you don't download the submodules?

**Reply by ramazulay** [reply]
Hi @D10S0VSkY-OSS,
You are right, My fault, thanks.
